stan/math/prim/fun/reverse.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef STAN_MATH_PRIM_FUN_REVERSE_HPP
+#define STAN_MATH_PRIM_FUN_REVERSE_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/vectorize/apply_vector_unary.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return a copy of the specified vector, row vector or array
+ * in reversed order.
+ *
+ * This implicitly allows for reverse with nested arrays (e.g.
```

Comment:
This needs to be changed so that it's not vectorized.  The function `reverse({a1, a2, a3})` should always return `{a3, a2, a1}` no matter what the types are.  That is, it should *not* return `{reverse(a1), reverse(a2), reverse(a3)}`.  If you want that, define a different function.

Reply:
I've reverted the change, so it has the behaviour you describe.